libs/driver/include/driver/VideoDriver.h

```diff
@@ -28,9 +28,16 @@ class VideoDriver : public IVideoDriver
     bool GetMouseStateR() const override;
 
     VideoMode GetWindowSize() const override final { return windowSize_; }
-    Extent GetRenderSize() const override final { return renderSize_; }
+    Extent GetRenderSize() const override final { return viewRenderSize_; }
```

Comment:
No. This is all by careful design. The render size in view space coordinates is the one users are interested in. It is scaled by GUI scale, which is fundamental to implementing the scaling effect. `glViewport` and `glScissor` are set to the window size, and `glOrtho` uses render size (as returned here). The original render size is only needed for internal calculations.

`viewRenderSize_` has to be updated by the backend in the `onGuiScaleChanged` callback. These steps are ordered this way to avoid spurious hover events. Arguably, I'm complicating things unnecessarily to avoid a minor visual glitch, but is carefully considered. :man_shrugging:

Reply:
> The render size in view space coordinates is the one users are interested in. It is scaled by GUI scale, which is fundamental to implementing the scaling effect.

So if the guiScale chantes the `viewRenderSize_` needs to change too, doesn't it?

> `viewRenderSize_` has to be updated by the backend in the `onGuiScaleChanged` callback.

I don't see this happening in this PR, am I missing it? Also `viewRenderSize_` is a property of this class so isn't this class responsible for keeping the values in sync?

Reply:
Dang it. My original response didn't make it.

Long story short. It is happening. But the fact that this causes confusion makes me want to refactor it.
The code has organically evolved by nudging existing functions to do new things, and it's time to rethink this part of the code from the ground up instead of shoe-horning it.

Reply:
Also, thinking about terminology, `viewRenderSize_` and `screenRenderSize_` don't make sense. All render sizes are always in screen coordinates. I should use `renderSize_` and `scaledRenderSize_` or similar.

Or, more accurately, the render size isn't even measured in pixels but points. But no need to make things more confusing.

Reply:
The easiest solution is to stop trying to be clever and duplicate the calculations in `SetNewSize()` and `setGuiScale()`. Both functions need things done slightly differently, so implementing one via the other is more complicated.

Reply:
Not sure I can follow. Where is `viewRenderSize_` updated? I only see `viewRenderSize_ = guiScale_.screenToView<Extent>(renderSize);` in `SetNewSize`

Would it make sense to simply not store/cache `viewRenderSize_` and calculate it on-the-fly in `GetRenderSize`?

Or even the other way round: `screenRenderSize_` is only used internally isn't it? Maybe rename that to `unscaledRenderSize_` or `logicalRenderSize_`(?) or don't store it at all and calculate from `viewRenderSize_` when required.   
On the other hand: When changing the scale you'd need to rescale `viewRenderSize_` anyway, so might as well store the "requested" (logical?) render size.

Damn, that stuff gets confusing ;-)

Reply:
It used to be updated this way: `onGuiScaleChanged()` -> `SDL2Video::UpdateCurrentSizes()` -> `VideoDriver::SetNewSize()`

Anyway, I've simplified it already and am handling a previously unhandled case (GUI scale re-calculates if DPI changes). Let's see what you think.

Caching it does make sense IMO, as it's quite frequently used.